src/commands/utility/my-points.js

```diff
@@ -38,103 +141,41 @@ module.exports = {
 
       const fetchedPoints = {
         taskCompleted: 0,
         addPoint: 0,
         boostedPoint: 0,
       };
-      function calculatePoints(messages, user, tagIds, fetchedPoints) {
-        messages.forEach((message) => {
-          if (message.author.id !== interaction.client.user.id) {
-            return;
-          }
-
-          const userMentionRegex = /<@(\d+)>/g;
-          const mentionedUsers = Array.from(
-            message.content.matchAll(userMentionRegex),
-            (match) => match[1]
-          );
-
-          mentionedUsers.forEach((mentionedUserId) => {
-            if (mentionedUserId === user.id) {
-              if (
-                message.content.includes(`<@&${tagIds.taskCompletedTagId}>`)
-              ) {
-                fetchedPoints.taskCompleted += 1;
-              }
-              if (message.content.includes(`<@&${tagIds.addPointTagId}>`)) {
-                fetchedPoints.addPoint += 1;
-              }
-              if (message.content.includes(`<@&${tagIds.boostedPointTagId}>`)) {
-                fetchedPoints.boostedPoint += 1;
-              }
-            }
-          });
-        });
-      }
-
-      async function processChannel(channelName) {
-        const channel = interaction.guild.channels.cache.find(
-          (ch) => ch.name === channelName && ch.isTextBased()
-        );
-        if (!channel) {
-          return;
-        }
-
-        const activeThreads = await channel.threads.fetchActive();
-        for (const thread of activeThreads.threads.values()) {
-          const threadCreationDate = new Date(thread.createdAt);
-          if (
-            threadCreationDate >= targetStartDate &&
-            threadCreationDate <= targetEndDate
-          ) {
-            const messages = await thread.messages.fetch();
-            calculatePoints(messages, user, tagIds, fetchedPoints);
-          }
-        }
-      }
 
       // Process channels concurrently
-      await Promise.all(channels.map(processChannel));
-
-      const monthName = new Intl.DateTimeFormat(interaction.locale, {
-        month: 'long',
-      }).format(new Date(year, month - 1));
-      const responseContent = [
-        `${user.tag} - ${monthName.charAt(0).toUpperCase() + monthName.slice(1)} - ${year}`,
-        `${translateLanguage('myPoints.taskComplete')}: ${fetchedPoints.taskCompleted || 0}`,
-        `${translateLanguage('myPoints.addPoints')}: ${fetchedPoints.addPoint || 0}`,
-        `${translateLanguage('myPoints.boostedPoints')}: ${fetchedPoints.boostedPoint || 0}`,
-        `${translateLanguage('myPoints.totalPoints')}: ${fetchedPoints.addPoint + fetchedPoints.boostedPoint || 0}`,
-      ].join('\n');
+      await Promise.all(
+        channels.map((channelName) =>
+          processChannel({
+            interaction,
+            channelName,
+            user,
+            fetchedPoints,
+            year,
+            month,
+          })
+        )
+      );
```

Comment:
I haven't noticed that  it modifies the fetchedPoints var and then it's used below. It's pretty confuse. But maybe we can work in it as a refactor later, since @ARenzDev is also modifying this logic and we don't want to have a lot of conflicts 😅